### PR TITLE
[SPARK-33630][SQL] Support SHOW TABLES command as table valued function

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Support `SHOW TABLES` command as a table valued function.
So user can be easy to reach metadata system information use SQL and analysis directory.

### Why are the changes needed?
user can be easy to reach metadata system information use SQL and analysis directory.

### Does this PR introduce _any_ user-facing change?
Yea, user can get table list by call table valued function `show_table()`
```
       sql("CREATE DATABASE d1")
        sql("CREATE TABLE t1(a INT) USING PARQUET")
        sql("CREATE TABLE d1.t1(a INT) USING PARQUET")
        sql("CREATE TABLE d1.t2(a INT) USING PARQUET")

        sql("SELECT * from show_tables()")
        sql("SELECT * from show_tables('default')")
        sql("SELECT * from show_tables('d1')")
        sql("SELECT * from show_tables('d1', 't1*')")
```

### How was this patch tested?
Added UT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
This PR now pending by https://github.com/apache/spark/pull/31245

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala [hunk 1]
```diff
@@ -2558,6 +2558,26 @@ abstract class SQLQuerySuiteBase extends QueryTest with SQLTestUtils with TestHi
     }
   }
 
+  test("SPARK-33630: Support SHOW command as table valued function") {
```
Comment (MaxGekk) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala:2561]: Please, place tests like this to the unified test suites for v1/v2 commands. See `ShowTablesSuite`.

**Comment by AngersZhuuuu** [comment]
retest this please

**Comment by maropu** [comment]
Is this related to https://github.com/apache/spark/pull/31548 ?

**Comment by AngersZhuuuu** [comment]
> Is this related to #31548 ?

Yea, hmmm since this is pending by https://issues.apache.org/jira/browse/SPARK-34157, so I left it until that pr merge.
Anyway it's a useful method.  Seems we use different way to do this. I think it’s good if you use either method.

**Pull request closed without merging**